AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2260,14 +2260,23 @@ - (void)displayWillStart
   [self _pendingNodeWillDisplay:self];
 
   [_supernode subnodeDisplayWillStart:self];
-
-  if (_placeholderImage && _placeholderLayer && self.layer.contents == nil) {
-    [CATransaction begin];
-    [CATransaction setDisableActions:YES];
-    [self _setupPlaceholderLayerContents];
-    _placeholderLayer.opacity = 1.0;
-    [CATransaction commit];
-    [self.layer addSublayer:_placeholderLayer];
+  
+  if (_placeholderEnabled && self.layer.contents == nil && [self __implementsDisplay]) {
+    if (!_placeholderImage) {
```

Comment:
This code is almost exactly correct, although doing it at this point Will mean that the sun elements which never become fully visible on screen may do work to re-create their placeholders as they are starting to display because of entering the off screen display range.

Because placeholders are synchronous, it is important to minimize the amount of work that it Is done to support them. Perhaps more importantly, in reviewing this I noticed a few other bugs, most notably that the network image node will remove its placeholder for the image has finished downloading because a display pass is allowed to complete on it even though it has no contents to draw.

Thank you for picking up this task and investigating, although I have started a new branch based off of yours to continue the change and posted it in a different pull request: https://github.com/facebook/AsyncDisplayKit/pull/1398